geode-core/src/test/java/org/apache/geode/cache/query/internal/ArithmeticOperationsIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+
+  private Cache cache;
+  QueryService queryService;
+
+  @Before
+  public void setup() {
+    String regionName = "portfolio";
+
+    cache = serverStarterRule.getCache();
+    assertNotNull(cache);
+    Region region =
+        cache.createRegionFactory().setDataPolicy(DataPolicy.REPLICATE).create(regionName);
+
+    Portfolio p = new Portfolio(1);
+    region.put(1, p);
+    Portfolio p2 = new Portfolio(2);
+    region.put(2, p2);
+    Portfolio p3 = new Portfolio(3);
+    region.put(3, p3);
+
+    queryService = cache.getQueryService();
+  }
+
+
+  @Test
+  public void modOnNonNumericShouldThrowException() throws Exception {
+    try {
+      queryService.newQuery("select * from /portfolio p where p % 2 = 1").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Arithmetic Operations can only be applied to numbers", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void modOnNullShouldThrowException() throws Exception {
+    try {
+      queryService.newQuery("select * from /portfolio p where null % 2 = 1").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Cannot evaluate arithmetic operations on null values", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void modOnNumericShouldApplyOperationAndReturnMatchingResults() throws Exception {
+    SelectResults modPercentageResults = (SelectResults) queryService
+        .newQuery("select * from /portfolio p where p.ID % 2 = 1").execute();
+
+    SelectResults modKeywordResults = (SelectResults) queryService
+        .newQuery("select * from /portfolio p where p.ID MOD 2 = 1").execute();
+
+    assertEquals(2, modPercentageResults.size());
+    assertEquals(modPercentageResults.size(), modKeywordResults.size());
+  }
+
+  @Test
+  public void modOperationAsProjectionShouldCorrectlyModReturnValue() throws Exception {
+    SelectResults modPercentageResults = (SelectResults) queryService
+        .newQuery("select (ID % 2) from /portfolio p where p.ID % 2 = 1").execute();
+
+    SelectResults modKeywordResults = (SelectResults) queryService
+        .newQuery("select ID mod 2 from /portfolio p where p.ID MOD 2 = 1").execute();
+
+    assertEquals(2, modPercentageResults.size());
+    assertEquals(modPercentageResults.size(), modKeywordResults.size());
+    for (Object o : modKeywordResults) {
+      assertEquals(1, o);
+    }
+  }
+
+  @Test
+  public void addOperationShouldApplyOpInQuery() throws Exception {
+    SelectResults results = (SelectResults) queryService
+        .newQuery("select * from /portfolio p where p.ID + 1 = 2").execute();
+    assertEquals(1, results.size());
+  }
+
+  @Test
+  public void addOperationShouldThrowExceptionWhenAddingNonNumeric() throws Exception {
+    try {
+      SelectResults results = (SelectResults) queryService
+          .newQuery("select * from /portfolio p where p.ID + '1' = 2").execute();
```

Comment:
tests with addition causing to overflow the int limit ?

Reply:
I think we will just use whatever java does with overflown values.  If we want to upcast after addition or multiplication then that can be a different feature?